superbench/benchmarks/micro_benchmarks/cuda_nccl_bw_performance.py

```diff
@@ -0,0 +1,139 @@
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class CudaNcclBwBenchmark(MicroBenchmarkWithInvoke):
+    """The NCCL bus bandwidth performance benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'all_reduce_perf'
+        self.__algorithms = {
+            'allreduce': 'all_reduce_perf',
+            'allgather': 'all_gather_perf',
+            'broadcast': 'broadcast_perf',
+            'reduce': 'reduce_perf',
+            'reducescatter': 'reduce_scatter_perf'
+        }
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--algo',
+            type=str,
+            nargs='+',
+            default=list(self.__algorithms.keys()),
+            help='Nccl algorithms to benchmark. E.g. {}.'.format(' '.join(list(self.__algorithms.keys()))),
+        )
+        self._parser.add_argument(
+            '--gpu_count',
+            type=int,
+            default=8,
+            help='The count of GPUs used for benchmarking.',
+        )
+        self._parser.add_argument(
+            '--max_size',
+            type=str,
+            default='8192M',
+            help='Max size in bytes to run the nccl test. E.g. 8192M.',
+        )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        # Format the arguments
+        if not isinstance(self._args.algo, list):
+            self._args.algo = [self._args.algo]
+        self._args.algo = [p.lower() for p in self._args.algo]
+
+        # Check the arguments and generate the commands
+        for algo in self._args.algo:
+            if algo not in self.__algorithms:
+                self._result.set_return_code(ReturnCode.INVALID_ARGUMENT)
+                logger.error(
+                    'Unsupported algorithm of NCCL test - benchmark: {}, algorithm: {}, expected: {}.'.format(
+                        self._name, algo, ' '.join(list(self.__algorithms.keys()))
+                    )
+                )
+                return False
+            else:
+                self._bin_name = self.__algorithms[algo]
+                if not self._set_binary_path():
+                    return False
+
+                command = os.path.join(self._args.bin_dir, self._bin_name)
+                command += ' -b 1 -e {} -f 2 -g {} -c 0'.format(self._args.max_size, self._args.gpu_count)
+                self._commands.append(command)
+
+        return True
+
+    def _process_raw_result(self, cmd_idx, raw_output):
```

Comment:
Haven't added the support for different rank?

Reply:
For single node 8 gpu mpirun, it can support. For multi-node, I do not have the environment to test currently.
![image](https://user-images.githubusercontent.com/37182275/125878744-130baad7-8e7b-4434-a401-ace97b88f14b.png)